pkg/controller/certificaterequests/sync.go

```diff
@@ -46,22 +47,23 @@ func (c *Controller) Sync(ctx context.Context, cr *v1alpha1.CertificateRequest)
 		return nil
 	}
 
-	if apiutil.CertificateRequestHasCondition(cr, v1alpha1.CertificateRequestCondition{
-		Type:   v1alpha1.CertificateRequestConditionReady,
-		Status: v1alpha1.ConditionFalse,
-		Reason: v1alpha1.CertificateRequestReasonFailed,
-	}) {
+	if apiutil.CertificateRequestHasFailed(cr) {
 		dbg.Info("certificate request condition failed so skipping processing")
 		return nil
 	}
 
 	crCopy := cr.DeepCopy()
+
 	defer func() {
 		if _, saveErr := c.updateCertificateRequestStatus(ctx, cr, crCopy); saveErr != nil {
 			err = utilerrors.NewAggregate([]error{saveErr, err})
 		}
 	}()
 
+	// If the CertificateRequest has the conditon 'Failed' then set the
+	// FailureTime to `c.clock.Now()`
+	defer c.setFailureTime(crCopy)
```

Comment:
Instead of including this here, can we include it as part of the `updateCertificateRequestStatus` func?

Alternatively, if we simply set the FailureTime at the point where we set the `reason` to `Failed`, we can be sure that both are recorded at the same time anyway (i.e. in the Recorder).

Reply:
(now that I've thought about it, the recorder probably *is* the most sensible place to set this.. not sure what you think?)